### PR TITLE
A few agent and installer updates

### PR DESCRIPTION
**Pull request by adjackura**
setup metadata scripts startup and shutdown run			
auto_updater no longer is keyed of metadata flag		
add logging to the event log

### CODE REVIEW

File: logger/logger.go [hunk 1]
```diff
@@ -13,47 +13,42 @@
...
 // the callers main(). Log functions can be called before Init(), but log
 // output will go to COM1.
 func Init(name, port string) {
 	logger = name
 	out := &serialPort{port}
+	// Split logging to the serial port and event log so processes like the
+	// metadata script runnner can log to serial output but not the event log.
 	Log = log.New(out, "", log.Ldate|log.Ltime)
+	if err := slSetup(name); err != nil {
+		log.Fatal(err)
```
Comment (ning-yang) [inline logger/logger.go]: why not use Log here. the default log goes to stdout and are not easy to see if running as a service.

File: logger/logger.go [hunk 2]
```diff
@@ -13,47 +13,42 @@
...
+	"strings"
 
 	"github.com/tarm/serial"
+	"golang.org/x/sys/windows/svc/eventlog"
 )
 
 var (
-	// Log is the logger's log.Logger.
+	// Log is the serial logger's log.Logger.
 	Log         *log.Logger
```
Comment (ning-yang) [inline logger/logger.go]: should we rename the Log to something like serialLogger so it is not a public var and more readable

File: logger/logger_windows.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	switch w.pri {
+	case sInfo:
+		return len(b), w.el.Info(1, string(b))
+	case sError:
+		return len(b), w.el.Error(2, string(b))
+	}
+	return 0, fmt.Errorf("unrecognized severity: %v", w.pri)
+}
+
+func newW(pri severity, src string) (*writer, error) {
```
Comment (ning-yang) [inline logger/logger_windows.go]: is this some naming conversion or we can have a more specific name

Comment (adjackura) [inline logger/logger_windows.go]: this is private so I'm not so concerned, the W stands for writer

**Pull request merged**